ctpolicy/applepolicy.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package ctpolicy contains structs describing CT policy requirements and
+// corresponding logic.
+package ctpolicy
+
+import (
+	"github.com/google/certificate-transparency-go/loglist"
+	"github.com/google/certificate-transparency-go/x509"
+)
+
+// AppleCTPolicy manages policy logic in appliance to the data provided.
+type AppleCTPolicy struct {
+	baseBuilder baseGroupBuilder
+}
+
+// LogsByGroup describes submission requirements according to https://support.apple.com/en-us/HT205280
+func (appleP *AppleCTPolicy) LogsByGroup(cert *x509.Certificate, approved *loglist.LogList) (map[string]*LogGroupInfo, error) {
+	var outerror error
+	groups := make(map[string]*LogGroupInfo)
+
+	baseGroup := BuildGroup(appleP.baseBuilder, approved)
+	switch m := lifetimeInMonths(cert); {
+	case m < 15:
+		if err := baseGroup.setMinInclusions(2); err != nil {
+			outerror = err
+		}
+	case m <= 27:
+		if err := baseGroup.setMinInclusions(3); err != nil {
+			outerror = err
+		}
+	case m <= 39:
+		if err := baseGroup.setMinInclusions(4); err != nil {
+			outerror = err
+		}
+	default:
+		if err := baseGroup.setMinInclusions(5); err != nil {
+			outerror = err
+		}
+	}
```

Comment:
Probably neater to pull all the error returns below:
```golang
	var err error
	switch m := lifetimeInMonths(cert); {
	case m < 15:
		err = baseGroup.setMinInclusions(2)
	case m <= 27:
		err = baseGroup.setMinInclusions(3)
	case m <= 39:
		err = baseGroup.setMinInclusions(4)
	default:
		err = baseGroup.setMinInclusions(5)
	}
	if err != nil {
		return nil, err
	}
```
or maybe:
```golang
	var incCount int
	switch m := lifetimeInMonths(cert); {
	case m < 15:
		incCount = 2
	case m <= 27:
		incCount = 3
	case m <= 39:
		incCount = 4
	default:
		incCount = 5
	}
	if err := baseGroup.setMinInclusion(incCount); err != nil {
		return nil, err
	}
```

Reply:
Picking second option.